### PR TITLE
Extensions

### PR DESCRIPTION
**Pull request by chfast**
- Fixes documentation.
- Add some helpers for EVMC and loader modules.

### CODE REVIEW

File: lib/loader/loader.c [hunk 1]
```diff
@@ -121,3 +122,26 @@ evmc_create_fn evmc_load(const char* filename, enum evmc_loader_error_code* erro
         *error_code = ec;
     return create_fn;
 }
+
+struct evmc_instance* evmc_load_and_create(const char* filename, enum evmc_loader_error_code* error_code)
+{
+    evmc_create_fn create_fn = evmc_load(filename, error_code);
```
Comment (axic) [inline lib/loader/loader.c]: I think it should be safe (at least on POSIX) to call this multiple times and not pollute the memory (since `dlopen` should just return the reference if it was loaded already IIRC). However, is that the case for Windows?

Do we need to maintain our local list of loaded EMVC libraries in order to make this helper callable multiple times?

**Comment by axic** [comment]
I don't think the go bindings file belong to this PR.

File: lib/loader/loader.c (hunk 1 above)
Comment (chfast) [reply lib/loader/loader.c]: It's the same of Windows.

>  Calling LoadLibrary increments the reference count.

**Comment by chfast** [comment]
> I don't think the go bindings file belong to this PR.

Removed.

File: include/evmc/helpers.h [hunk 2]
```diff
@@ -0,0 +1,52 @@
+/* EVMC: Ethereum Client-VM Connector API.
+ * Copyright 2018 Pawel Bylica.
+ * Licensed under the MIT License. See the LICENSE file.
+ */
+
+/**
+ * EVMC Helpers
+ *
+ * A collection of helper functions for invoking a VM instance methods.
+ * These are convenient e.g. for Go bindings, because this is the only way to invoke a function
```
Comment (axic) [inline include/evmc/helpers.h]: Would reword that to something like: `These are convenient for languages where invoking foreign runtime loaded functions is "ugly" or problematic (such as Go).`

File: include/evmc/helpers.h [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+#pragma once
+
+#include <evmc/evmc.h>
+
+/**
+ * Destroys the VM instance.
+ *
+ * @see evmc_destroy_fn
+ */
+inline void evmc_destroy(struct evmc_instance* instance)
```
Comment (axic) [inline include/evmc/helpers.h]: These should be `static inline` even though we have `#pragma once`. Not even sure we should have pragma once since that is more of a C++ism.

File: include/evmc/instructions.h [hunk 4]
```diff
@@ -3,13 +3,19 @@
...
+ *
+ * @defgroup instructions EVM Instructions
+ * @{
+ */
 #pragma once
 
 #include <evmc/evmc.h>
 #include <evmc/utils.h>
 
-#include <stdint.h>
```
Comment (axic) [inline include/evmc/instructions.h]: Why remove this? `int16_t` still requires it.

File: include/evmc/loader.h [hunk 5]
```diff
@@ -48,18 +70,31 @@ enum evmc_loader_error_code
...
- *                    ::EVMC_ERRC_SUCCESS on success or any other error code as described above.
+ *                    ::EVMC_LOADER_SUCCESS on success or any other error code as described above.
  * @return            The pointer to the EVM create function or NULL.
  */
 evmc_create_fn evmc_load(const char* filename, enum evmc_loader_error_code* error_code);
 
+/**
+ * Dynamically loads the VM DLL and creates the VM instance.
+ *
+ * This is a macro of creating the VM instance with the function returned from evmc_load().
```
Comment (axic) [inline include/evmc/loader.h]: `macro for`

File: include/evmc/helpers.h (hunk 3 above)
Comment (chfast) [reply include/evmc/helpers.h]: `#pragma once` is not C++. Is... pragma. All compilers support it.

File: lib/loader/loader.c (hunk 1 above)
Comment (axic) [reply lib/loader/loader.c]: It should be documented then that calling `evmc_load` and `evmc_load_and_create` multiple times should be safe and not blow up memory usage.

File: include/evmc/instructions.h (hunk 4 above)
Comment (chfast) [reply include/evmc/instructions.h]: It's already included in `<evmc/evmc.h>`.

File: include/evmc/helpers.h (hunk 3 above)
Comment (axic) [reply include/evmc/helpers.h]: Still, if we target FFI in languages which may not support pragma, can we just have `static` here since this in a header?
